### PR TITLE
Clamp latency inside pre-defined limits

### PR DESCRIPTION
**Pull request by achronop**
Create redefine limit and clamp latency inside. If more than one stream is initialized limits only lower values of latency are allowed.

### CODE REVIEW

File: src/cubeb_audiounit.cpp [hunk 1]
```diff
@@ -62,8 +62,8 @@ typedef UInt32  AudioFormatFlags;
 //#define LOGGING_ENABLED
 #ifdef LOGGING_ENABLED
 #define LOG(...) do {                           \
+    fprintf(stderr, "(line: %d) ", __LINE__);   \
     fprintf(stderr, __VA_ARGS__);               \
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: This remove the \n, so it kind of break logging. Maybe you can put a third printf that only prints a \n.

File: src/cubeb_audiounit.cpp [hunk 2]
```diff
@@ -1014,6 +1015,70 @@ audiounit_destroy_input_linear_buffer(cubeb_stream * stream)
   delete stream->input_linear_buffer;
 }
 
+static uint32_t
+audiounit_clamp_latency(cubeb_stream * stm,
+                              uint32_t latency_frames)
+{
+  // For the 1st stream set anything within safe min-max
+  if (stm->context->active_streams == 1) {
```
Comment (padenot) [inline src/cubeb_audiounit.cpp]: This has to be called after incrementing the stream count. Maybe we can just `assert(stm->context->active_streams != 0)` to catch this.

File: src/cubeb_audiounit.cpp (hunk 1 above)
Comment (achronop) [inline src/cubeb_audiounit.cpp]: I did it that way in order to print the line and then the message in one line. It looks fine in my tests. I you do not like it I can bring it back.

Comment (padenot) [inline src/cubeb_audiounit.cpp]: But then we have all the logging on one line, right ?

Comment (achronop) [inline src/cubeb_audiounit.cpp]: I see your point with endl, many `LOG(...)` still use endl. They should be removed I guess.

**Pull request merged**